engine/src/core/graphics/rendering/rendering_device/batch_2d.cpp

```diff
@@ -0,0 +1,158 @@
+//
+// Created by blinku on 23.11.2019.
+//
+
+#include "batch_2d.h"
+#include "graphics/rendering/camera_2d.h"
+#include "graphics/rendering/sprite.h"
+#include <cstring>
+
+namespace Bald::Graphics {
+    static constexpr uint32_t MAX_SPRITES = 10000;
+    static constexpr uint32_t SPRITE_VERTEX_DATA_SIZE = sizeof(SpriteVertexData);
+    static constexpr uint32_t SPRITE_SIZE = 4 * SPRITE_VERTEX_DATA_SIZE;
+    static constexpr uint32_t INDICES_SIZE = 6;
+    static constexpr uint32_t MAX_VERTICES = SPRITE_SIZE * MAX_SPRITES;
+    static constexpr uint32_t MAX_INDICES = INDICES_SIZE * MAX_SPRITES;
+
+    Batch2D::Batch2D() :
+        m_MaxVertices(MAX_VERTICES),
```

Comment:
if those are global constants why do we need to hold them as members?

Reply:
We might want to have them altered in some way in the future. Also, currently we support single batching, but it could get expanded and accessing members would then be the only way to get specific values from the batches (batches are not stateless).